proposals/0106-paritioned-epoch-rewards-pda.md

```diff
@@ -0,0 +1,100 @@
+---
+simd: '0106'
+title: Epoch Rewards Partition Data PDA
+authors:
+  - Tyera Eulberg
+category: Standard
+type: Core
+status: Draft
+created: 2024-01-17
+feature: https://github.com/solana-labs/solana/issues/32166
+extends: 0015
+---
+
+## Summary
+
+Partitioned epoch rewards (SIMD 0015) will split epoch rewards distribution
+across multiple blocks at the start of an epoch. This extension of that SIMD
+proposes storing data about the rewards calculation and partitions in a sysvar
+account at a program-derived address (PDA), which can be queried by clients, or
+indeed within the runtime, to performantly identify the partition for the
+distribution to any particular address.
+
+## Motivation
+
+When we move to partitioned epoch rewards as described in the original SIMD, the
+only way to find the stake or voting rewards for a specific address will be to
+iterate through blocks at the beginning of each epoch, inspecting all the
+rewards. This is because the runtime does not persist information about how the
+rewards were partitioned; in fact, it does not even persist how many blocks the
+rewards distribution spans, so there is no way to predict how long it will take
+or how far through the epoch to go in finding a particular address.
+
+## Alternatives Considered
+
+An alternative to storing partition data in an on-chain account would be to
+record the necessary data in the ledger in some fashion. This could be a
+transaction that gets added to the block, a special RewardType, or metadata that
+gets stored on the node and then duplicated to long-term storage, like Bigtable.
+In fact, it is probably worthwhile to pursue this alternative as well, since it
+will enable finding rewards without access to snapshots or the running chain.
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+When partitioned rewards are calculated in the runtime (currently during the
```

Comment:
We should be more specific about when in the first block this account is created.

Reply:
I can add that, but I feel that's really something that should be defined in the [original partitioned-rewards SIMD](https://github.com/solana-foundation/solana-improvement-documents/blob/main/proposals/0015-partitioned-epoch-reward-distribution.md), if it needs to be specified in a SIMD. I feel it is sufficient and correct for this SIMD to say that the sysvar population happens as part of the Rewards Calculation operation (whenever that might be).

Reply:
I see, I was specifically just concerned that the combination of the two SIMD's make it clear that the PDA should be created by the runtime before transactions are processed. Some of the wording had felt a little loose and ambiguous. I think your change makes it more clear, thanks!

Reply:
I just read SIMD 15 and it specifically says that the sysvar is created at the beginning of the first block before processing transactions. Nitpicky but I think it would be nice if we are similarly clear about that here and say that this PDA sysvar is created at the same time as that one.